tests/004-streams.cpp

```diff
@@ -209,47 +212,40 @@ namespace oxen::quic::test
 
         REQUIRE(tls_future.get());
 
-        // 1) open 12 streams and send
-        for (int i = 0; i < 12; ++i)
+        for (int i = 0; i < n_streams; ++i)
         {
-            std::thread stream_thread([&]() {
+            client_endpoint->call([&]() {
                 streams[i] = conn_interface->get_new_stream();
                 streams[i]->send(msg);
-                // set send promises
                 send_promises[i].set_value(true);
             });
-            stream_thread.join();
         }
 
         // 2) check the first 8
-        for (int i = 0; i < 8; ++i)
+        for (int i = 0; i < n_streams - 4; ++i)
             REQUIRE(receive_futures[i].get());
 
-        // 3) close 4 streams
+        // 3) close 5 streams
         for (int i = 0; i < 5; ++i)
-        {
-            std::thread close_thread([&]() { streams[i]->close(); });
-            close_thread.join();
-        }
+            streams[i]->close();
 
         // 4) check the last 4
-        for (int i = 8; i < 12; ++i)
+        for (int i = n_streams - 4; i < n_streams; ++i)
             REQUIRE(receive_futures[i].get());
 
         // 5) open 2 more streams and send
         for (int i = 0; i < 2; ++i)
         {
-            std::thread open_thread([&]() {
+            client_endpoint->call([&]() {
```

Comment:
A `call` here shouldn't be needed (and, in fact, if we need it, then something is wrong in the API); we ought to be able to just directly call the code here, from the main thread.